StripePaymentSheet/StripePaymentSheet/Source/Helpers/PaymentSheetLinkAccount.swift

```diff
@@ -238,19 +236,19 @@ class PaymentSheetLinkAccount: PaymentSheetLinkAccountInfoProtocol {
     func createLinkAccountSession(
         completion: @escaping (Result<LinkAccountSession, Error>) -> Void
     ) {
-        guard let session = currentSession else {
-            stpAssertionFailure()
-            completion(
-                .failure(
-                    PaymentSheetError.unknown(
-                        debugDescription: "Linking account session without valid consumer session"
+        retryingOnAuthError(completion: completion) { [publishableKey] completionWrapper in
+            guard let session = self.currentSession else {
+                stpAssertionFailure()
+                completion(
```

Comment:
This code is getting harder to read, because it's unclear when to call the completion vs the completionWrapper. I think your code is correct, but do you have any thoughts on how to make it more clear?

Reply:
Agreed, let me try making this more obvious. I haven't really changed this code, I'm just moving it around.

Reply:
Okay, renamed the variable to make it clearer. `completion` calls the completion block, while `completionRetryingOnAuthErrors` is the wrapped completion that checks for auth errors before completing.